test/e2e/scrapeconfig_test.go

```diff
@@ -3197,3 +3200,549 @@ var OpenStackSDTestCases = []scrapeCRDTestCase{
+						Key: "key.pem",
+					},
+				},
+			},
+		},
+		expectedError: true,
+	},
+	{
+		name: "Valid Role - Instance",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+			ScalewaySDConfigs: []monitoringv1alpha1.ScalewaySDConfig{
+				{
+					ProjectID: "1",
+					Role:      monitoringv1alpha1.ScalewayRoleInstance,
+					AccessKey: "AccessKey",
+					SecretKey: v1.SecretKeySelector{
+						LocalObjectReference: v1.LocalObjectReference{
+							Name: "secret",
+						},
+						Key: "key.pem",
+					},
+				},
+			},
+		},
+		expectedError: false,
+	},
+	{
+		name: "Valid Role - Baremetal",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+			ScalewaySDConfigs: []monitoringv1alpha1.ScalewaySDConfig{
+				{
+					ProjectID: "1",
+					Role:      monitoringv1alpha1.ScalewayRoleBaremetal,
+					AccessKey: "AccessKey",
+					SecretKey: v1.SecretKeySelector{
+						LocalObjectReference: v1.LocalObjectReference{
+							Name: "secret",
+						},
+						Key: "key.pem",
+					},
+				},
+			},
+		},
+		expectedError: false,
+	},
+	{
+		name: "Invalid Role",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+			ScalewaySDConfigs: []monitoringv1alpha1.ScalewaySDConfig{
+				{
+					ProjectID: "1",
+					Role:      "Invalid Role",
+					AccessKey: "AccessKey",
+					SecretKey: v1.SecretKeySelector{
+						LocalObjectReference: v1.LocalObjectReference{
+							Name: "secret",
+						},
+						Key: "key.pem",
+					},
+				},
+			},
+		},
+		expectedError: true,
+	},
+	{
+		name: "Missing Role",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+			ScalewaySDConfigs: []monitoringv1alpha1.ScalewaySDConfig{
+				{
+					ProjectID: "1",
+					AccessKey: "AccessKey",
+					SecretKey: v1.SecretKeySelector{
+						LocalObjectReference: v1.LocalObjectReference{
+							Name: "secret",
+						},
+						Key: "key.pem",
+					},
+				},
+			},
+		},
+		expectedError: true,
+	},
+	{
+		name: "Valid Api Url",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+			ScalewaySDConfigs: []monitoringv1alpha1.ScalewaySDConfig{
+				{
+					ProjectID: "1",
+					Role:      monitoringv1alpha1.ScalewayRoleInstance,
+					AccessKey: "AccessKey",
+					SecretKey: v1.SecretKeySelector{
+						LocalObjectReference: v1.LocalObjectReference{
+							Name: "secret",
+						},
+						Key: "key.pem",
+					},
+					ApiURL: ptr.To("https://api.scaleway.com"),
+				},
+			},
+		},
+		expectedError: false,
+	},
+	{
+		name: "Invalid Api Url",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+			ScalewaySDConfigs: []monitoringv1alpha1.ScalewaySDConfig{
+				{
+					ProjectID: "1",
+					Role:      monitoringv1alpha1.ScalewayRoleInstance,
+					AccessKey: "AccessKey",
+					SecretKey: v1.SecretKeySelector{
+						LocalObjectReference: v1.LocalObjectReference{
+							Name: "secret",
+						},
+						Key: "key.pem",
+					},
+					ApiURL: ptr.To("ftp://example.com"),
+				},
+			},
+		},
+		expectedError: true,
+	},
+	{
+		name: "Missing Api Url",
```

Comment:
Hi @nutmos this are a bit duplicated as a ApiURL is a optional field, so the condition in this test is the same as foe example the "Valid Project ID". Makes sense to remove.

Reply:
I wouldn't mind keeping it just for the sake of completeness.

Reply:
Ok them thanks